### PR TITLE
Add eval function in Tpsa class

### PR DESCRIPTION
**Pull request by xresende**

### CODE REVIEW

**Review by murilobalves — APPROVED** [review]

File: include/trackcpp/tpsa.h [hunk 1]
```diff
@@ -464,12 +476,30 @@ void Tpsa<V,N,TYPE>::get_power(unsigned int idx_, unsigned int* power_) {
     }
 }
 
+template <unsigned int V, unsigned int N, typename TYPE>
+TYPE Tpsa<V,N,TYPE>::eval(const std::vector<TYPE>& val) const {
+    if (val.size() != V) {
+        std::cerr << "Invalid vector size in Tpsa::eval!" << std::endl;
```
Comment (fernandohds564) [inline include/trackcpp/tpsa.h:482]: why not throw an error here?

Comment (xresende) [reply include/trackcpp/tpsa.h:482]: to keep the standard. we have now used the throw/catch mechanism yet in trackcpp. but we can create a new PR for this. I think it is better than sending all exceptions to stderr...

Comment (fernandohds564) [reply include/trackcpp/tpsa.h:482]: I think I'm starting to use the throw mechanism in my PR, but I don't know the behavior when using it in python. Does swig intercepts this and raise a python exception?

Comment (xresende) [reply include/trackcpp/tpsa.h:482]: dunno

Comment (fernandohds564) [reply include/trackcpp/tpsa.h:482]: I looked into it (asked ChatGPT) and yes, there is a way to configure SWIG to wrap C++ exceptions into python exceptions. I'm implementing it in my PR on the IDs pass methods. With that in mind, I think throwing exceptions is a good way to handle problems in C++.

**Review by VitorSouzaLNLS — APPROVED** [review]

File: include/trackcpp/tpsa.h (hunk 1 above)
Comment (xresende) [reply include/trackcpp/tpsa.h:482]: @fernandohds564 , can you commit a change here using exception instead of cerr ?

Comment (fernandohds564) [reply include/trackcpp/tpsa.h:482]: Maybe we could leave this kind of change to another PR, what do you think? I think there is a lot of places where we would benefit from this type of change.

**Review by fernandohds564 — APPROVED** [review]

**Pull request merged**